logical/sanitize.go

```diff
@@ -60,3 +65,46 @@ type HTTPWrapInfo struct {
 	CreationTime    time.Time `json:"creation_time"`
 	WrappedAccessor string    `json:"wrapped_accessor,omitempty"`
 }
+
+type HTTPSysInjector struct {
+	Response *HTTPResponse
+}
+
+func (h HTTPSysInjector) MarshalJSON() ([]byte, error) {
+	j, err := json.Marshal(h.Response)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fast path no data or empty data
+	if h.Response.Data == nil || len(h.Response.Data) == 0 {
+		return j, nil
+	}
+
+	// Marshaling a response will always be a JSON object, meaning it will
+	// always start with '{', so we hijack this to prepend necessary values
+
+	// Make a guess at the capacity, and write the object opener
+	buf := bytes.NewBuffer(make([]byte, 0, len(j)+len(h.Response.Data)*100))
+	buf.WriteRune('{')
+
+	for k, v := range h.Response.Data {
+		// Marshal each key/value individually
+		mk, err := json.Marshal(k)
```

Comment:
No, because the string may have values that have to be escaped (for instance, newlines or quotes). Marshaling ensures that everything is properly escaped.